src/main/java/org/jenkinsci/plugins/githubautostatus/GithubBuildStatusGraphListener.java

```diff
@@ -300,10 +302,37 @@ private static boolean isDeclarativePipelineJob(FlowNode fn) {
         ArrayList<String> result = new ArrayList<>();
         for (ModelASTStage stage : modelList) {
             result.add(stage.getName());
+            result.addAll(getNestedStages(stage.toJSON()));
         }
         return result;
     }
 
+    /**
+     * Returns a list of the nested stages within a stage's JSON object.
+     * If we were using pipeline-model-api >= 1.2.8 we could use the getStages()
```

Comment:
I wonder if we should move the dependency up to 1.2.8 and use those methods? I think it might be safer. If you don't want to do that now, we could add an issues.

Also, it would be nice to see some tests for this method.